simulator/kernel/mars/mars_kernel.cpp

```diff
@@ -228,9 +233,32 @@ void MARSKernel::read_from_file() {
     mem->memcpy_host_to_guest( buffer_ptr, byte_cast( buffer.data()), chars_to_read);
 }
 
-void MARSKernel::connect_memory( std::shared_ptr<FuncMemory> m)
+void MARSKernel::connect_riscv_handler()
+{
+    constexpr Addr TRAP_VECTOR = 0x8'000'0000;
+    sim->write_csr_register( "stvec", TRAP_VECTOR);
+    auto pc = trap_vector_address<Addr>( TRAP_VECTOR);
+    ElfLoader elf_loader( KERNEL_IMAGES "riscv32.bin");
+    elf_loader.load_to( mem.get(), pc - elf_loader.get_text_section_addr());
+}
+
+void MARSKernel::connect_mars_handler()
 {
-    BaseKernel::connect_memory( m);
     ElfLoader elf_loader( KERNEL_IMAGES "mars32_le.bin");
     elf_loader.load_to( mem.get(), 0x8'0000'0180 - elf_loader.get_text_section_addr());
 }
+
+void MARSKernel::connect_exception_handler()
+{
+    auto isa = sim->get_isa();
+    if ( isa == "riscv32") 
+    {
+        connect_riscv_handler();
+    } else if ( isa == "mars" || isa == "mips32le" || isa == "mips32") 
+    {
+        connect_mars_handler();
+    } else 
+    {
```

Comment:
Brackets are redundant:
https://github.com/MIPT-ILab/mipt-mips/wiki/Code-style-guidelines#pythonesque-statements

We follow the guideline to move "then/else" blocks to named functions more often.

Actually even "if"s are bad style, we should use polymorphism instead. No need to change it now, but if complexity increases, we'll switch it.